src/components/protocol_handler/src/handshake_handler.cc

```diff
@@ -0,0 +1,197 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "protocol_handler/handshake_handler.h"
+
+#include <bson_object.h>
+
+#include "protocol_handler/protocol_handler_impl.h"
+#include "protocol_handler/protocol_packet.h"
+#include "security_manager/security_manager.h"
+#include "protocol_handler/session_observer.h"
+
+namespace protocol_handler {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "ProtocolHandler")
+
+// DEPRECATED
+HandshakeHandler::HandshakeHandler(
+    ProtocolHandlerImpl& protocol_handler,
+    SessionObserver& session_observer,
+    uint32_t connection_key,
+    ConnectionID connection_id,
+    uint8_t session_id,
+    uint8_t protocol_version,
+    uint32_t hash_id,
+    ServiceType service_type,
+    const std::vector<int>& force_protected_service,
+    const bool is_new_service,
+    ProtocolPacket::ProtocolVersion& full_version,
+    std::shared_ptr<uint8_t> payload)
+    : protocol_handler_(protocol_handler)
+    , session_observer_(session_observer)
+    , context_()
+    , full_version_(full_version)
+    , protocol_version_(protocol_version)
+    , payload_(payload) {}
+
+HandshakeHandler::HandshakeHandler(
+    ProtocolHandlerImpl& protocol_handler,
+    SessionObserver& session_observer,
+    ProtocolPacket::ProtocolVersion& full_version,
+    const StartingSessionContext& context,
+    const uint8_t protocol_version,
+    std::shared_ptr<uint8_t> payload)
+    : protocol_handler_(protocol_handler)
+    , session_observer_(session_observer)
+    , context_(context)
+    , full_version_(full_version)
+    , protocol_version_(protocol_version)
+    , payload_(payload) {}
+
+HandshakeHandler::~HandshakeHandler() {
+  LOG4CXX_DEBUG(logger_, "Destroying StartSessionHandler: " << this);
+}
+
+uint32_t HandshakeHandler::connection_key() const {
+  return session_observer_.KeyFromPair(context_.connection_id_,
+                                       context_.new_session_id_);
+}
+
+bool HandshakeHandler::GetPolicyCertificateData(std::string& data) {
+  return false;
+}
+
+void HandshakeHandler::OnCertificateUpdateRequired() {}
+
+bool HandshakeHandler::OnHandshakeDone(
+    uint32_t connection_key,
+    security_manager::SSLContext::HandshakeResult result) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (connection_key != this->connection_key()) {
+    LOG4CXX_DEBUG(logger_, "Notification not for handler " << this);
+    return false;
+  }
+
+  const bool success =
+      result == security_manager::SSLContext::Handshake_Result_Success;
+
+  BsonObject params;
+  if (payload_) {
+    params = bson_object_from_bytes(payload_.get());
+  } else {
+    bson_object_initialize_default(&params);
+  }
+
+  if (success) {
+    ProcessSuccessfulHandshake(connection_key, params);
+  } else {
+    ProcessFailedHandshake(params);
+  }
+
+  bson_object_deinitialize(&params);
+  return true;
```

Comment:
@AKalinich-Luxoft Do we need to return always ` true` ? Is it really necessary?

Reply:
@AByzhynar yes. See my answer above:
https://github.com/smartdevicelink/sdl_core/pull/1971#discussion_r158717782